File: Programming Challenges/Max Profit/Max Profit.cs
// Imports
using System;
using System.IO;
using System.Linq;

class Program {
  // Function that generates an int array of random integers
  static int[] numArrayGen() {
      int[] data = new int[10000];
      Random rmd = new Random();
      for (int i = 0; i < 10000; i++) {
        data[i] = rmd.Next(1, 20000);
      }
      return data;
  }

  // Function that takes an array of ints and calculates the max profit
  static int maxProfit(int[] data) {
    int[] filteredArray = data.Skip(Array.IndexOf(data, data.Min())).ToArray();
    int profit = filteredArray.Max() - data.Min();
    return profit;
  }

  // Main Function
  static void Main(string[] args) {
    int[] data = numArrayGen();
    Console.WriteLine(maxProfit(data));
  }
}

File: Programming Challenges/Max Profit/Max Profit.py
# Import(s)
import random

# Function that takes a list of data and calculates the most profit that can be made
def maxProfit(data: list):
    filteredData = list(data[data.index(min(data)):]);
    profit = int(max(filteredData) - min(data));
    return profit;

# Testing the function
data = list([]);
for i in range(10000):
    data.append(random.randint(1, 50001));
print(maxProfit(data));